Add SUSE Edge 3.5.2 release notes and documentation updates (#1185)
suse-edge/suse-edge.github.io: asciidoc/day2/migration.adoc

  --namespace longhorn-system \
  --docker-server=dp.apps.rancher.io \
  --docker-username="${APPS.RANCHER.IO_USERNAME}" \
  --docker-password="${APPS.RANCHER.IO_ACCESS_TOKEN}"
----
+

. Finally, upgrade your Longhorn installation to SUSE Storage:
+
[,bash,subs="attributes"]
----
helm upgrade longhorn oci://dp.apps.rancher.io/charts/suse-storage \
	--namespace longhorn-system \
	--version {version-longhorn-chart} \
	--set privateRegistry.registrySecret=rancher-app-collection
----
+
You can provide a `values.yaml` file by appending `-f values.yaml` to the upgrade command if you wish.

==== Prepare for Rancher Turtles upgrade

[NOTE]
====
Applies only to CAPI/Metal3 management clusters that require a rancher turtles chart upgrade.

You must ensure the management cluster is first updated to the latest 3.4 z-stream, which contains the necessary 0.24.3 Rancher Turtles version.
====

Starting with Rancher 2.13, Rancher Turtles is installed by default, therefore it is necessary to follow some additional migration steps as described in the https://documentation.suse.com/cloudnative/cluster-api/latest/en/tutorials/migration.html[Rancher Turtles Documentation]

> Before upgrade it is necessary to remove the installed rancher-turtles chart and rancher-turtles-airgap-resources (if installed),

Reviewer: ```suggestion
Before the upgrade, you must remove the installed `rancher-turtles` chart and `rancher-turtles-airgap-resources` (if present).
```

kubectl delete -n kube-system helmchart rancher-turtles-airgap-resources
----

Next we must patch the CRD resources as described in the https://turtles.docs.rancher.com/turtles/next/en/tutorials/migration.html[Rancher Turtles Documentation]

[,bash]
----
kubectl patch crd capiproviders.turtles-capi.cattle.io --type=json -p='[{"op": "add", "path": "/metadata/annotations/meta.helm.sh~1release-namespace", "value": "cattle-turtles-system"}]'
kubectl patch crd clusterctlconfigs.turtles-capi.cattle.io --type=json -p='[{"op": "add", "path": "/metadata/annotations/meta.helm.sh~1release-namespace", "value": "cattle-turtles-system"}]'
----

Now follow the regular steps to upgrade the management cluster to `Edge {static-edge-version}`

==== Rancher Turtles post-upgrade

*After* following the steps below to upgrade to `Edge {static-edge-version}` it is necessary to install the new `rancher-turtles-providers` helm chart - this creates new `CAPIProvider` resources to replace those removed in the pre-upgrade steps above.

First we must run the migration script as described in the https://turtles.docs.rancher.com/turtles/next/en/tutorials/migration.html[Rancher Turtles Documentation]

> 

Reviewer: ```suggestion
First, we must run the migration script as described in the https://turtles.docs.rancher.com/turtles/next/en/tutorials/migration.html[Rancher Turtles Documentation]: 
```

```
wget https://raw.githubusercontent.com/rancher/turtles/refs/heads/release/v0.25/scripts/migrate-providers-ownership.sh
bash migrate-providers-ownership.sh --adopt metal3:capm3-system --adopt metal3ipam:metal3-ipam-system
```

> Next the `rancher-turtles-providers` chart can be installed.  This chart installation should be done via a `HelmChart` resource to enable future automated upgrade via the upgrade controller:

Reviewer: ```suggestion
Next, install the `rancher-turtles-providers` chart using a HelmChart resource, which enables future automated upgrades via the upgrade controller.
```
